libevmasm/Assembly.cpp

```diff
@@ -500,7 +500,7 @@ map<u256, u256> Assembly::optimiseInternal(
 	return tagReplacements;
 }
 
-LinkerObject const& Assembly::assemble() const
+LinkerObject const& Assembly::assemble(ErrorReporter& _errorReporter) const
 {
 	// Return the already assembled object, if present.
 	if (!m_assembledObject.bytecode.empty())
```

Comment:
If we assemble for the first time, we need `ErrorReporter`. If we return a cached object, we don't.

If we have sufficient amount of cases where we do know that the cashed object already exists, we may implement `cachedAssemblyObject()` without any arguments.